Error compiling esp8266

With an example config:
```
esphome:
  name: climatest
  friendly_name: climatest

esp8266:
  board: d1_mini

logger:
api:
ota:
wifi:
  ssid: !secret ssid
  password: !secret PWD

external_components:
  source: github://khenderick/esphome-opentherm
  components: [opentherm]

opentherm:
  read_pin: 4
  write_pin: 5
**The rest is from Example entry for config.yaml:**
``` 
I have this compilation error:
```
INFO ESPHome 2023.5.5
INFO Reading configuration /config/climatest.yaml...
INFO Generating C++ source...
INFO Compiling app...
Processing climatest (board: d1_mini; framework: arduino; platform: platformio/espressif8266@3.2.0)
--------------------------------------------------------------------------------
HARDWARE: ESP8266 80MHz, 80KB RAM, 4MB Flash
Dependency Graph
|-- ESPAsyncTCP-esphome @ 1.2.3
|-- ESPAsyncWebServer-esphome @ 2.1.0
|   |-- ESPAsyncTCP-esphome @ 1.2.3
|   |-- Hash @ 1.0
|   |-- ESP8266WiFi @ 1.0
|-- DNSServer @ 1.1.1
|-- ESP8266WiFi @ 1.0
|-- ESP8266mDNS @ 1.2
|-- noise-c @ 0.1.4
|   |-- libsodium @ 1.10018.1
Compiling .pioenvs/climatest/src/esphome/components/opentherm/number/custom_number.cpp.o
Compiling .pioenvs/climatest/src/esphome/components/ota/ota_backend_arduino_esp8266.cpp.o
Compiling .pioenvs/climatest/src/esphome/components/ota/ota_backend_arduino_rp2040.cpp.o
Compiling .pioenvs/climatest/src/esphome/components/ota/ota_backend_esp_idf.cpp.o
Compiling .pioenvs/climatest/src/esphome/components/ota/ota_component.cpp.o
Compiling .pioenvs/climatest/src/esphome/components/sensor/automation.cpp.o
In file included from src/esphome/components/opentherm/number/custom_number.cpp:1:
src/esphome/components/opentherm/number/custom_number.cpp: In member function 'void esphome::opentherm::CustomNumber::dump_custom_config(const char*, const char*)':
src/esphome/components/opentherm/number/custom_number.cpp:31:22: error: initializer fails to determine size of '__pstr__'
   31 |   LOG_NUMBER(prefix, type, this);
src/esphome/core/log.h:94:90: note: in definition of macro 'esph_log_config'
   94 |   esp_log_printf_(ESPHOME_LOG_LEVEL_CONFIG, tag, __LINE__, ESPHOME_LOG_FORMAT(format), ##__VA_ARGS__)
      |                                                                                          ^~~~~~~~~~~
src/esphome/components/number/number.h:14:5: note: in expansion of macro 'ESP_LOGCONFIG'
   14 |     ESP_LOGCONFIG(TAG, "%s%s '%s'", prefix, LOG_STR_LITERAL(type), (obj)->get_name().c_str()); \
      |     ^~~~~~~~~~~~~
src/esphome/core/log.h:185:28: note: in expansion of macro 'LOG_STR_ARG'
  185 | #define LOG_STR_LITERAL(s) LOG_STR_ARG(LOG_STR(s))
      |                            ^~~~~~~~~~~
src/esphome/core/log.h:185:40: note: in expansion of macro 'LOG_STR'
  185 | #define LOG_STR_LITERAL(s) LOG_STR_ARG(LOG_STR(s))
      |                                        ^~~~~~~
src/esphome/components/number/number.h:14:45: note: in expansion of macro 'LOG_STR_LITERAL'
   14 |     ESP_LOGCONFIG(TAG, "%s%s '%s'", prefix, LOG_STR_LITERAL(type), (obj)->get_name().c_str()); \
      |                                             ^~~~~~~~~~~~~~~
src/esphome/components/opentherm/number/custom_number.cpp:31:3: note: in expansion of macro 'LOG_NUMBER'
   31 |   LOG_NUMBER(prefix, type, this);
      |   ^~~~~~~~~~
src/esphome/components/opentherm/number/custom_number.cpp:31:22: error: array must be initialized with a brace-enclosed initializer
   31 |   LOG_NUMBER(prefix, type, this);
src/esphome/core/log.h:94:90: note: in definition of macro 'esph_log_config'
   94 |   esp_log_printf_(ESPHOME_LOG_LEVEL_CONFIG, tag, __LINE__, ESPHOME_LOG_FORMAT(format), ##__VA_ARGS__)
      |                                                                                          ^~~~~~~~~~~
src/esphome/components/number/number.h:14:5: note: in expansion of macro 'ESP_LOGCONFIG'
   14 |     ESP_LOGCONFIG(TAG, "%s%s '%s'", prefix, LOG_STR_LITERAL(type), (obj)->get_name().c_str()); \
      |     ^~~~~~~~~~~~~
src/esphome/core/log.h:185:28: note: in expansion of macro 'LOG_STR_ARG'
  185 | #define LOG_STR_LITERAL(s) LOG_STR_ARG(LOG_STR(s))
      |                            ^~~~~~~~~~~
src/esphome/core/log.h:185:40: note: in expansion of macro 'LOG_STR'
  185 | #define LOG_STR_LITERAL(s) LOG_STR_ARG(LOG_STR(s))
      |                                        ^~~~~~~
src/esphome/components/number/number.h:14:45: note: in expansion of macro 'LOG_STR_LITERAL'
   14 |     ESP_LOGCONFIG(TAG, "%s%s '%s'", prefix, LOG_STR_LITERAL(type), (obj)->get_name().c_str()); \
      |                                             ^~~~~~~~~~~~~~~
src/esphome/components/opentherm/number/custom_number.cpp:31:3: note: in expansion of macro 'LOG_NUMBER'
   31 |   LOG_NUMBER(prefix, type, this);
      |   ^~~~~~~~~~
*** [.pioenvs/climatest/src/esphome/components/opentherm/number/custom_number.cpp.o] Error 1
========================== [FAILED] Took 3.52 seconds ==========================
``` 
Hovewer esp32 compiles OK.

--- a/components/opentherm/number/custom_number.cpp
+++ b/components/opentherm/number/custom_number.cpp
@@ -27,8 +27,7 @@ void CustomNumber::control(float value) {
   }
 };
 
-void CustomNumber::dump_custom_config(const char *prefix, const char *type) {
-  LOG_NUMBER(prefix, type, this);
+void CustomNumber::dump_custom_config(const char *prefix) {
   if (!std::isnan(this->initial_value_)) {
     ESP_LOGCONFIG(TAG, "%s  Initial value: '%f'", prefix, this->initial_value_);
   }

--- a/components/opentherm/number/custom_number.h
+++ b/components/opentherm/number/custom_number.h
@@ -12,7 +12,7 @@ class CustomNumber : public Component, public number::Number {
   void set_restore_value(bool restore_value) { this->restore_value_ = restore_value; }
   void set_initial_value(float initial_value) { this->initial_value_ = initial_value; }
   void setup() override;
-  void dump_custom_config(const char *prefix, const char *type);
+  void dump_custom_config(const char *prefix);
 
  protected:
   bool restore_value_{false};

--- a/components/opentherm/opentherm.cpp
+++ b/components/opentherm/opentherm.cpp
@@ -146,10 +146,12 @@ void OpenThermComponent::dump_config() {
 #endif
 #ifdef USE_NUMBER
   if (this->ch_setpoint_temperature_number_) {
-    this->ch_setpoint_temperature_number_->dump_custom_config("  ", "CH setpoint temperature:");
+    LOG_NUMBER("  ", "CH setpoint temperature:", this->ch_setpoint_temperature_number_);
+    this->ch_setpoint_temperature_number_->dump_custom_config("  ");
   }
   if (this->dhw_setpoint_temperature_number_) {
-    this->dhw_setpoint_temperature_number_->dump_custom_config("  ", "DHW setpoint temperature:");
+    LOG_NUMBER("  ", "DHW setpoint temperature:", this->dhw_setpoint_temperature_number_);
+    this->dhw_setpoint_temperature_number_->dump_custom_config("  ");
   }
 #endif
 }

--- a/components/opentherm/opentherm.h
+++ b/components/opentherm/opentherm.h
@@ -1,5 +1,6 @@
 #pragma once
 
+#include "esphome/core/helpers.h"
 #include "esphome/core/defines.h"
 #include "esphome/core/component.h"
 #include "esphome/core/gpio.h"

